app/src/main/java/free/rm/skytube/businessobjects/YouTube/Tasks/GetYouTubeVideosTask.java

```diff
@@ -115,7 +122,26 @@ protected void onPreExecute() {
 
 	@Override
 	protected void onPostExecute(List<YouTubeVideo> videosList) {
-		videoGridAdapter.appendList(videosList);
+
+		try {
+
+			ArrayList<YouTubeVideo> youTubeVideoList = new ArrayList<>();
```

Comment:
The code that checks if a channel is blocked or not should be move to `doInBackground()` as we must never block the main/UI thread as otherwise the app will stutter.

Also there is no need for the `blockedChannelsDb` instance variable.  Just remove it and place it in the `doInBackground()` or call `BlockedChannelsDb.getBlockedChannelsDb()` directly.